iothub/device/tests/ClientPropertyCollectionTests.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class ClientPropertyCollectionTests
+    {
+        private const string BoolPropertyName = "boolProperty";
```

Comment:
I see some of these string values end 'Name', but some do not.

Reply:
Good catch, I'll fix it in a subsequent PR